openshift/openshift/src/main/java/org/arquillian/cube/openshift/impl/client/CubeOpenShiftConfiguration.java

```diff
@@ -285,11 +269,12 @@ public int getRouterSniPort() {
     @Override
     public String getToken() {
 
-        if ((token == null || token.isEmpty()) && (client != null)) {
-            token = client.getClientExt().getConfiguration().getOauthToken();
+        if ((super.getToken() == null || super.getToken().isEmpty()) && (client != null)) {
+          String token = client.getClientExt().getConfiguration().getOauthToken();
+          setToken(token);
         }
 
-        return token;
+        return super.getToken();
```

Comment:
Why `super.` here? Do we have any extra logic above to be concerned about?

Reply:
Using `super.` covers both scenario. i.e.
1. When there is no token, with client not null, it should get token from config and set it. and then return it.
2.  If token is available in parent, then return it.